core/rootobject/collection.go

```diff
@@ -0,0 +1,294 @@
+	// "ancestor" may or may not be nil.
+	HandleMerge(ctx context.Context, mro merge.MergeRootObject) (doltdb.RootObject, *merge.MergeStats, error)
+	// LoadCollection loads the Collection from the given root.
+	LoadCollection(ctx context.Context, root RootValue) (Collection, error)
+	// LoadCollectionHash loads the Collection hash from the given root.
+	LoadCollectionHash(ctx context.Context, root RootValue) (hash.Hash, error)
+	// PutCollection updates the Collection in the given root, returning the updated root.
+	PutCollection(ctx context.Context, root RootValue, coll Collection) (RootValue, error)
+	// Serializer returns the serializer associated with this Collection.
+	Serializer() RootObjectSerializer
+}
+
+// Collection is a collection of root objects.
+type Collection interface {
+	// DropRootObject removes the given root object from the collection.
+	DropRootObject(ctx context.Context, identifier id.Id) error
+	// GetID returns the identifying ID for the Collection.
+	GetID() RootObjectID
+	// GetRootObject returns the root object matching the given ID. Returns false if it cannot be found.
+	GetRootObject(ctx context.Context, identifier id.Id) (RootObject, bool, error)
+	// HasRootObject returns whether a root object exactly matching the given ID was found.
+	HasRootObject(ctx context.Context, identifier id.Id) (bool, error)
+	// IDToTableName converts the given ID to a table name. The table name will be empty for invalid IDs.
+	IDToTableName(identifier id.Id) doltdb.TableName
+	// IterateIDs iterates over all IDs in the Collection.
+	IterateIDs(ctx context.Context, callback func(identifier id.Id) (stop bool, err error)) error
+	// IterateRootObjects iterates over all root objects in the Collection.
+	IterateRootObjects(ctx context.Context, callback func(rootObj RootObject) (stop bool, err error)) error
+	// PutRootObject updates the Collection with the given root object. This may error if the root object already exists.
+	PutRootObject(ctx context.Context, rootObj RootObject) error
+	// RenameRootObject changes the ID for a root object matching the old ID.
+	RenameRootObject(ctx context.Context, oldID id.Id, newID id.Id) error
+	// ResolveName finds the closest matching (or exact) ID for the given name. If an exact match is not found, then
+	// this may error if the name is ambiguous.
+	ResolveName(ctx context.Context, name doltdb.TableName) (doltdb.TableName, id.Id, error)
+	// TableNameToID converts the given name to an ID. The ID will be invalid for empty/malformed names.
+	TableNameToID(name doltdb.TableName) id.Id
+}
+
+var (
+	// globalCollectionFunctions maps each ID to the CollectionFunctions.
+	globalCollectionFunctions = make([]CollectionFunctions, rootObjectID_NumberOfItems)
+	// globalCollectionInitialized tracks whether the functions have finished initializing.
+	globalCollectionInitialized = false
+)
+
+// RegisterCollection registers the given set of functions.
+func RegisterCollection(collection CollectionFunctions) {
```

Comment:
Would kill this, not migrate it to the new package. Having `AllCollections` (in the new package) call three (3) methods from three (3) packages is not a big deal, certainly not enough to justify this indirectino.